### PR TITLE
Improvements to backporting tool

### PR DESCRIPTION
**Pull request by aholstrup1**
#### Summary 
* If gh cli isn't installed, link to a winget command rather than a URL 
* Improved error message if branch doesn't exist
* Ability to run New-BCAppsBackport from outside your local BCApps repository folder 

#### Work Item(s) 
Fixes [AB#498430](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/498430)

### CODE REVIEW

File: build/scripts/CrossBranchPorting.psm1 [hunk 1]
```diff
@@ -21,81 +21,92 @@ function New-BCAppsBackport() {
...
-            # Create a new branch for the cherry-pick
-            $cherryPickBranch = "hotfix/$TargetBranch/$branchNameSuffix"
-
-            # Port the pull request to the target branch
-            PortPullRequest -PullRequestDetails $pullRequestDetails -TargetBranch $TargetBranch -CherryPickBranch $cherryPickBranch
-
-            # Create a pull request for the cherry-pick
-            gh pr create --title $title --body $body --base $TargetBranch --head $cherryPickBranch
+        # Go back to the original branch
+        RunAndCheck git checkout $startingBranch
```
Comment (mazhelez) [inline build/scripts/CrossBranchPorting.psm1]: This should be in a `finally` block.

**Review by mazhelez — APPROVED** [review]
Left a minor comment, but overall looks great.

**Review by mazhelez — DISMISSED** [review]

File: build/scripts/CrossBranchPorting.psm1 [hunk 2]
```diff
@@ -21,81 +21,92 @@ function New-BCAppsBackport() {
...
+    finally {
+        Pop-Location
     }
 }
 
 function PrecheckBackport($TargetBranches, $PullRequestNumber) {
     # Check gh cli is installed
     if (-not (Get-Command gh -ErrorAction SilentlyContinue)) {
-        throw "Please install the GitHub CLI from https://cli.github.com/"
+        Write-Host "Please install the GitHub CLI by running 'winget install --id GitHub.cli' from your terminal. Once installed open a new terminal and run 'gh auth login' to authenticate with GitHub." -ForegroundColor Red
```
Comment (darjoo) [inline build/scripts/CrossBranchPorting.psm1:108]: Is it possible to just run the command if the user gives consent?
Then says open a new terminal and run `gh auth login`. If that's more work, just leave it

**Review by darjoo — DISMISSED** [review]

File: build/scripts/CrossBranchPorting.psm1 (hunk 2 above)
Comment (aholstrup1) [reply build/scripts/CrossBranchPorting.psm1:108]: Considering the user has to close and reopen their shell, I'll leave running the winget command as a manual step for now. 

About `gh auth login`; It is probably a good idea to validate that gh cli is authenticated (and authenticate if it isn't) prior to starting the backport. I've added a check for the PrecheckBackport function

**Review by mazhelez — APPROVED** [review]

**Review by darjoo — APPROVED** [review]

**Pull request merged**